contributors/guide/README.md

```diff
@@ -157,7 +157,19 @@ _Improvements needed_
 
 ## Open a Pull Request
 
-PR workflow is described [here](https://git.k8s.io/community/contributors/devel/pull-requests.md#the-testing-and-merge-workflow).
+Pull requests are often called simply "PR".  This jargon in itself is a hurdle to newcomers, but the open source pull request model of development is widely documented and searchable.   Kubernetes generally follows the standard [github pull request](https://help.github.com/articles/about-pull-requests/) process, but there is a layer of additional kubernetes specific (and sometimes SIG specific) differences.
```

Comment:
I think we can leave out the second sentence in order to be brief.
I love the link to github pull request documentation.

Reply:
Will remove.